drivers/i2c/i2c_nrfx_twi_rtio.c

```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright (c) 2018, Nordic Semiconductor ASA
+ * Copyright (c) 2024, Croxel Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/i2c/rtio.h>
+#include <zephyr/dt-bindings/i2c/i2c.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <soc.h>
+#include <nrfx_twi.h>
+
+#include <zephyr/logging/log.h>
+#include <zephyr/irq.h>
+LOG_MODULE_REGISTER(i2c_nrfx_twi, CONFIG_I2C_LOG_LEVEL);
+
+#if CONFIG_I2C_NRFX_TRANSFER_TIMEOUT
+#define I2C_TRANSFER_TIMEOUT_MSEC K_MSEC(CONFIG_I2C_NRFX_TRANSFER_TIMEOUT)
+#else
+#define I2C_TRANSFER_TIMEOUT_MSEC K_FOREVER
+#endif
+
+struct i2c_nrfx_twi_data {
+	struct i2c_rtio *ctx;
+	bool twi_enabled;
+	uint32_t dev_config;
+};
+
+struct i2c_nrfx_twi_config {
+	nrfx_twi_t twi;
+	nrfx_twi_config_t config;
+	const struct pinctrl_dev_config *pcfg;
+};
```

Comment:
It would be great to have code shared between this driver and `i2c_nrfx_twi.c` extracted to a separate file.
From what I see that would be config, most of `i2c_nrfx_twi_msg_start`, `i2c_nrfx_twi_configure`, `i2c_nrfx_twi_recover_bus`, `init_twi`, `twi_nrfx_pm_action` and most of device defines.

Reply:
Request addressed. The following APIs have been refactored:
- `struct i2c_nrfx_twi_config`
- `init_twi` -> `i2c_nrfx_twi_init`
- `i2c_nrfx_twi_configure`
- `i2c_nrfx_twi_recover_bus`
- `twi_nrfx_pm_action`
- Body of `i2c_nrfx_twi_msg_start` -> `i2c_nrfx_twi_msg_transfer`
- Device Macros.